src/main/kotlin/com/sourcegraph/cody/chat/ui/ContextFileActionLink.kt

```diff
@@ -0,0 +1,60 @@
+package com.sourcegraph.cody.chat.ui
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.project.Project
+import com.intellij.ui.JBColor
+import com.intellij.ui.components.AnActionLink
+import com.sourcegraph.cody.agent.protocol.ContextFileFile
+import java.awt.Color
+import java.awt.Graphics
+
+class ContextFileActionLink(
+    project: Project,
+    contextFileFile: ContextFileFile,
+    anAction: AnAction
+) : AnActionLink("", anAction) {
+  private val localFileBackground = JBColor(Color(182, 210, 242), Color(56, 85, 112))
+  private val isReferringToLocalFile = contextFileFile.repoName == null
+
+  init {
+    text = contextFileFile.getLinkActionText(project.basePath)
+    toolTipText = contextFileFile.uri.path
+  }
+
+  override fun paintComponent(g: Graphics) {
+    if (isReferringToLocalFile) {
+      g.color = localFileBackground
+
+      val fm = g.fontMetrics
+      val rect = fm.getStringBounds(text, g)
+      val textWidth = rect.width.toInt()
+      g.fillRect(0, 0, textWidth, height)
+    }
+    super.paintComponent(g)
+  }
+
+  private fun ContextFileFile.getLinkActionText(projectPath: String?): String {
```

Comment:
Any reason why we cannot move this to `ContextFileFile`?

Reply:
Also, if you can move it there it would be great to also add some tests for it.

Reply:
for me it sounds more logical to keep this method here - closer to the scope of it's usage (`ContextFileActionLink` class). `ContextFileFile` file is responsible only for data type definition (and serialisation).

Reply:
Either way please add the tests :)

Reply:
done!